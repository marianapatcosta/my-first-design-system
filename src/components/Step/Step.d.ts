import React, { CSSProperties, ReactNode } from 'react';
export interface StepProps {
    /**
     * element id
     */
    id?: string;
    /**
     * additional css class
     */
    className?: string;
    /**
     * additional style
     */
    style?: CSSProperties;
    /**
     * step content
     */
    content: string | ReactNode;
    /**
     * Is this the current step?
     */
    isActive: boolean;
}
export declare const Step: React.FC<StepProps>;
