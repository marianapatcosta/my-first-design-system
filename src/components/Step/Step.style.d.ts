export declare const StyledStep: import("styled-components").StyledComponent<"div", any, {
    isActive?: boolean | undefined;
}, never>;
