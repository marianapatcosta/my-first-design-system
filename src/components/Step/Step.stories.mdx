import { Meta } from '@storybook/addon-docs/blocks';
import { Step } from './Step';

<Meta title='Design System|Step' component={Step} />

# Step
Step component

## Usage
Use step component as children of Stepper; used to show a part of Stepper's content, divided in smaller parts.

- Default Step:
  <Story name='Default'>
    <Step />
  </Story>

## Other important notes
not applicable.
