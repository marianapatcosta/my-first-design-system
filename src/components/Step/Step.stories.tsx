import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Step, StepProps } from './Step';

export default {
  title: 'Step',
  component: Step,
  args: {},
} as Meta;

const Template: Story<StepProps> = (args) => <Step {...args} />;

export const Default = Template.bind({});
Default.args = { content: <div>Step </div>, isActive: true };
