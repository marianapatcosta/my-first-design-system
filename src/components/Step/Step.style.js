var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
import { baseStyles } from '../../styles';
export var StyledStep = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", ";\n  width: 100%;\n\n  ", ";\n"], ["\n  ", ";\n  width: 100%;\n\n  ", ";\n"])), baseStyles, function (_a) {
    var isActive = _a.isActive;
    return !isActive && "display: none";
});
var templateObject_1;
