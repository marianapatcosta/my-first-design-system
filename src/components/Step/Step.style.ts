import styled from 'styled-components';
import { baseStyles } from '../../styles';

export const StyledStep = styled.div<{ isActive?: boolean }>`
  ${baseStyles};
  width: 100%;

  ${({isActive}) => !isActive && `display: none`};
`;
