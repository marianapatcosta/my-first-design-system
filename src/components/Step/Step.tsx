import React, { CSSProperties, ReactNode } from 'react';
import { StyledStep } from './Step.style';

export interface StepProps {
  /**
   * element id
   */
  id?: string;
  /**
   * additional css class
   */
  className?: string;
  /**
   * additional style
   */
  style?: CSSProperties;
  /**
   * step content
   */
  content: string | ReactNode;
  /**
   * Is this the current step?
   */
  isActive: boolean;
}

export const Step: React.FC<StepProps> = ({ content, isActive, ...props }) => {
  return (
    <StyledStep {...props} isActive={isActive}>
      {content}
    </StyledStep>
  );
};

Step.defaultProps = {
  id: '',
  className: '',
  style: {},
  content: '',
  isActive: false
};
