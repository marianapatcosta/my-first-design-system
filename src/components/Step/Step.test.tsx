import React from 'react';
import { render } from '@testing-library/react';
import { Step } from './Step';

describe('Step', () => {
  const defaultProps = { content: <div>Step </div>, isActive: true };
  test('renders without errors and matches snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Step {...defaultProps} />);

    expect(firstChild).toMatchSnapshot();
  });

  test('should display content', () => {
    const { getByText } = render(<Step {...defaultProps} />);
    getByText('Step');
  });

  test('should not display content if isActive is false', () => {
    const { getByText } = render(<Step {...defaultProps} isActive={false} />);
    !getByText('Step');
  });
});
