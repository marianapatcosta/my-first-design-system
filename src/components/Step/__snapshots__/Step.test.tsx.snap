// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step renders without errors and matches snapshot 1`] = `
<div
  class="sc-bdfBwQ kcDtUs"
  id=""
>
  <div>
    Step 
  </div>
</div>
`;
